.circleci/config.yml

```diff
@@ -1,26 +1,52 @@
-# CircleCI 2.0 configuration file. See <https://circleci.com/docs/2.0/language-python/>.
 version: 2
 jobs:
-  test_notebooks:
-    working_directory: ~/tutorial/notebooks
-
+  deps:
+    working_directory: ~/tutorial
     docker:
       - image: python:3.7
 
     steps:
       - checkout:
           path: ~/tutorial
 
+      - run:
+          name: Create a virtualenv
+          command: python -m venv /tmp/venv/tutorial
+
       - run:
           name: Install dependencies
-          command: make install
+          command: |
+            source /tmp/venv/tutorial/bin/activate
+            make install
+
+      - persist_to_workspace:
+          root: /tmp/venv/tutorial
+          paths:
+            - "*"
+
+  test_notebooks:
+    working_directory: ~/tutorial/notebooks
+    docker:
+      - image: python:3.7
+
+    steps:
+      - checkout:
+          path: ~/tutorial
+
+      - attach_workspace:
+          at: /tmp/venv/tutorial
 
       - run:
           name: Test notebooks
-          command: make test
+          command: |
+            source /tmp/venv/tutorial/bin/activate
+            make test
 
 workflows:
   version: 2
-  test:
+  build:
```

Comment:
I'm not sure I understand why two separate jobs and a venv are required.

Reply:
Because we're changing of working directory.

Reply:
I'm still not sure this additional complexity is buying us something that we need right now, and it carries a risk specific to the notebooks repo: we intend to build a particular workflow (the stable/edge thing) that we have not used before, and a simpler starting point would make that less error-prone. So I'd tend to apply YAGNI in this particular situation.

Reply:
The reason of adding a `virtualenv` is precisely to mitigate that risk.

> A simpler starting point would make that less error-prone

What does it exactly means simpler in this context? For me simple means reducing complexity, that is, the amount of things that I don't know and that I can't manage. And that's exactly what I'm doing here: I don't know what exactly comes in a Docker image, but I know the `virtualenv` contract and I trust it.

Also, how by that definition you mean to make the workflow less error-prone? I have the impression you're implying that less LOC means simpler... is that it?

I'm not sure of understanding what are you asking me to do and why.

Reply:
Removed `virtualenv` and reduced jobs.

Interesting discussion but not very time-effective. GTM? 😃

Reply:
I'd tend to prefer a single-job workflow as a starting point for the next step (stable/edge jobs). I'll defer to @sandcha for reviewing this however, as my understanding is that she's leading that and I'm serving as backup.

"Simple" for me tends to mean the [XP design rules](https://martinfowler.com/bliki/BeckDesignRules.html).

Reply:
> Interesting discussion but not very time-effective.

It takes time to communicate, and gain mutual understanding. This has been open less than 24h, so by the guidelines we've agreed there's no rush to bring it to a close.

I understand that it would have taken less time to simply request changes with a comment "this should just be one job". I did not have the level of certainty required to do that. The other option is to not review at all and stay silent, but this runs the risk of PRs getting stale by apathy, as well as that of not using the insights of everyone on the team.

Reply:
> It takes time to communicate

Hi @Morendil, what I mean is that it is an interesting discussion, but not very time-effective *for me*: for this particular case (`circleci.yml`), I'm better off making a compromise than reaching consensus. 

> I'll defer to @sandcha for reviewing this however

@sandcha is off for a week.

Reply:
(for me simple means simple as in Cynefin)